### PR TITLE
fix: Refactor Azure Unused Volume and Unused IP Actions

### PR DESCRIPTION
**Pull request by bryankaraffa**
### Description

- Refactored the `Delete Volume` and `Create Snapshot`, and `Delete IP Address` Actions to use updated Azure APIs, improve debugging, and error handling

### Link to Example Applied Policy

https://app.flexera.com/orgs/27018/automation/applied-policies/projects/116186?policyId=650304aa3a1bb5000199025d

Tested actions with single volume, multiple volumes, and volumes that have delete locks (expect error)

### Contribution Check List

- [X] New functionality includes testing.
- [X] New functionality has been documented in the README if applicable
- [X] New functionality has been documented in CHANGELOG.MD

### CODE REVIEW

File: cost/azure/unused_volumes/azure_unused_volumes.pt [hunk 1]
```diff
@@ -774,88 +775,84 @@ end
...
-      call sys_log("In for item =",to_s($item))
       if $param_take_snapshot == "true"
-        call create_snapshot($item) retrieve $status_code
-        if to_s($status_code)=="202"
+        call create_snapshot($item) retrieve $response
+        # We don't expect to get anything other than Succeeded, but just in case before deleting disk lets check
+        if $response["body"]["properties"]["provisioningState"] == "Succeeded"
           call delete_unattached_disk($item['id'])
         else
-          call sys_log("Azure create snapshot error statusCode: ",to_s($status_code))
```
Comment (XOmniverse) [inline cost/azure/unused_volumes/azure_unused_volumes.pt:791]: Was this intentionally removed? If so, should the above else statement still be there?

**Review by XOmniverse — CHANGES_REQUESTED** [review]
Only one minor thing that I spotted.

File: cost/azure/unused_volumes/azure_unused_volumes.pt (hunk 1 above)
Comment (bryankaraffa) [reply cost/azure/unused_volumes/azure_unused_volumes.pt:791]: Fixed in 667b048.  good catch!

**Comment by bryankaraffa** [comment]
@flexera-public/policy-developers  -- if someone has some time this week for another review please

**Review by XOmniverse — APPROVED** [review]
LGTM

**Pull request merged**